Розрахуй 95% довірчого інтервалу для середнього значення кількості проданого товару.
Інтерпретуй отриманий інтервал довіри: що це означає для бізнес-рішень?

In [6]:
import numpy as np
import scipy.stats as stats

sales = np.array([5, 15, 7, 9, 2, 10, 11, 25, 7, 8, 2, 3, 10, 10, 8])

mean_sales = np.mean(sales)
print("середнє значення", mean_sales)

std_dev = np.std(sales, ddof=1)
print("стандартне відхилення", mean_sales)

n = len(sales)
print("кількість елементів у вибірці", n)

середнє значення 8.8
стандартне відхилення 8.8
кількість елементів у вибірці 15


In [7]:
confidence = 0.95
confidence_interval = stats.t.interval(
    confidence, df=n - 1, loc=mean_sales, scale=std_dev / np.sqrt(n)
)

print(f"Довірчий інтервал: {confidence_interval}")

Довірчий інтервал: (np.float64(5.6311858948214155), np.float64(11.968814105178586))


Результат показав, що з ймовірністю 95% справжнє середнє значення витрат клієнтів знаходиться у межах від 5.63 до 11.96 одиниць.